### ISSUE TITLE
Vue-tsc translate some Vue compilation outputs

### ISSUE THREAD

**Issue by dizzyn** [johnsoncodehk/volar#1484]
Please help. I have some problem with all versions upper than "0.32.1" in my case it is "0.38.1". When I hit "vue-tsc --noEmit" it will try to validate Vue transformations outputs: 

![image](https://user-images.githubusercontent.com/2071816/174597381-89a0906d-0ac2-428d-97b5-3bd714865d6b.png)

My setup:
- vite 2.9.12
- vue 3.2.37
- typescript 4.7.4
- tsconfig:

````
{
  "compilerOptions": {
    "target": "esnext",
    "useDefineForClassFields": true,
    "module": "esnext",
    "moduleResolution": "node",
    "strict": true,
    "jsx": "preserve",
    "sourceMap": true,
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "lib": ["esnext", "dom"],
    "skipLibCheck": true,
    "types": ["vite/client"]
  },
  "include": [
    "**/*.ts",
    "**/*.d.ts",
    "src/**/*.tsx",
    "src/**/*.vue",
    "src/**/*.graphql",
    "test/**/*.ts",
    "test/**/*.d.ts",
    "test/**/*.vue"
  ],
  "references": [{ "path": "./tsconfig.node.json" }]
}
````

**Comment by johnsoncodehk** [comment]
Hi @dizzyn, this problem should be fixed in [v0.37.9](https://github.com/johnsoncodehk/volar/blob/master/CHANGELOG.md#0379-2022616), please make sure you have update vue-tsc version to latest in the project.

If you're, please let me know if you can reproduce this problem in https://github.com/johnsoncodehk/volar-starter or not.

**Comment by dizzyn** [comment]
Hi, I have reproduced that on the starter but it looks that the bug was on my side :)

1. remove   "allowJs": true, 
2. add JS (not TS) Vue component

So thank you and have a nice day
